src/topology/continuous_on.lean

```diff
@@ -710,84 +716,110 @@ begin
       { exact (hg x hx).congr
           (λ y hy, piecewise_eq_of_not_mem _ _ _ hy.2)
           (piecewise_eq_of_not_mem _ _ _ hx.2) } } }
 end
 
-lemma continuous_on_if {α β : Type*} [topological_space α] [topological_space β] {p : α → Prop}
-  {h : ∀a, decidable (p a)} {s : set α} {f g : α → β}
+lemma continuous_on.if' {s : set α} {p : α → Prop} {f g : α → β} [∀ a, decidable (p a)]
+  (hpf : ∀ a ∈ s ∩ frontier {a | p a},
+    tendsto f (𝓝[s ∩ {a | p a}] a) (𝓝 $ if p a then f a else g a))
+  (hpg : ∀ a ∈ s ∩ frontier {a | p a},
+    tendsto g (𝓝[s ∩ {a | ¬p a}] a) (𝓝 $ if p a then f a else g a))
+  (hf : continuous_on f $ s ∩ {a | p a}) (hg : continuous_on g $ s ∩ {a | ¬p a}) :
+  continuous_on (λ a, if p a then f a else g a) s :=
+hf.piecewise' hpf hpg hg
+
+lemma continuous_on.if {α β : Type*} [topological_space α] [topological_space β] {p : α → Prop}
+  [∀ a, decidable (p a)] {s : set α} {f g : α → β}
   (hp : ∀ a ∈ s ∩ frontier {a | p a}, f a = g a) (hf : continuous_on f $ s ∩ closure {a | p a})
   (hg : continuous_on g $ s ∩ closure {a | ¬ p a}) :
   continuous_on (λa, if p a then f a else g a) s :=
 begin
-  apply continuous_on_if',
+  apply continuous_on.if',
   { rintros a ha,
     simp only [← hp a ha, if_t_t],
     apply tendsto_nhds_within_mono_left (inter_subset_inter_right s subset_closure),
-    exact (hf a ⟨ha.1, ha.2.1⟩).tendsto },
+    exact hf a ⟨ha.1, ha.2.1⟩ },
   { rintros a ha,
     simp only [hp a ha, if_t_t],
     apply tendsto_nhds_within_mono_left (inter_subset_inter_right s subset_closure),
     rcases ha with ⟨has, ⟨_, ha⟩⟩,
     rw [← mem_compl_iff, ← closure_compl] at ha,
-    apply (hg a ⟨has, ha⟩).tendsto, },
+    apply hg a ⟨has, ha⟩ },
   { exact hf.mono (inter_subset_inter_right s subset_closure) },
   { exact hg.mono (inter_subset_inter_right s subset_closure) }
 end
 
-lemma continuous_if' {p : α → Prop} {f g : α → β} {h : ∀a, decidable (p a)}
-  (hpf : ∀ a ∈ frontier {x | p x},
-    tendsto f (nhds_within a {x | p x}) (𝓝 $ ite (p a) (f a) (g a)))
-  (hpg : ∀ a ∈ frontier {x | p x},
-    tendsto g (nhds_within a {x | ¬p x}) (𝓝 $ ite (p a) (f a) (g a)))
+lemma continuous_on.piecewise {s t : set α} {f g : α → β} [∀ a, decidable (a ∈ t)]
+  (ht : ∀ a ∈ s ∩ frontier t, f a = g a) (hf : continuous_on f $ s ∩ closure t)
+  (hg : continuous_on g $ s ∩ closure tᶜ) :
+  continuous_on (piecewise t f g) s :=
+hf.if ht hg
+
+lemma continuous_if' {p : α → Prop} {f g : α → β} [∀ a, decidable (p a)]
+  (hpf : ∀ a ∈ frontier {x | p x}, tendsto f (𝓝[{x | p x}] a) (𝓝 $ ite (p a) (f a) (g a)))
+  (hpg : ∀ a ∈ frontier {x | p x}, tendsto g (𝓝[{x | ¬p x}] a) (𝓝 $ ite (p a) (f a) (g a)))
   (hf : continuous_on f {x | p x}) (hg : continuous_on g {x | ¬p x}) :
   continuous (λ a, ite (p a) (f a) (g a)) :=
 begin
   rw continuous_iff_continuous_on_univ,
-  apply continuous_on_if'; simp; assumption
+  apply continuous_on.if'; simp *; assumption
+end
+
+lemma continuous_if {p : α → Prop} {f g : α → β} [∀ a, decidable (p a)]
```

Comment:
I agree that this lemma and the next one are the right ones. However, in applications you often have global continuity assumptions. So, registering `continuous.if` and `continuous.piecewise` seems like a good idea. (I have the impression that they could be useful in your modifications to `topology/path_connected.lean`, for instance).

Reply:
Done.